src/figurl/useSyncHive.ts

```diff
@@ -0,0 +1,135 @@
+import { JSONStringifyDeterministic } from "kachery-js/types/kacheryTypes"
+import { useCallback, useMemo, useState } from "react"
+
+// The "sync hive" is where we keep track of the state that is synchronized between figures.
+
+// Hive contains all the value/setValue pairs associated with sync IDs.
+// There should be only one Hive instance per browser session.
+// This is instantiated inside the useSyncHive() hook below.
+class Hive {
+    // all the value/setValue pairs associated with sync IDs
+    #items: {[key: string]: {value: any, setValue: (a: any)=> void}} = {}
+    // The constructor takes an onChange() callback
+    constructor(private onChange: () => void) {
+    }
```

Comment:
What's the significance of a body-less constructor in JS? Does it automatically store the parameter in an instance variable or something?

Reply:
yes, with the private keyword the onChange becomes a private member.